test/tests/nested_em.text

```diff
@@ -1,3 +1,15 @@
 *test **test** test*
 
 _test __test__ test_
+
+_test_ test _test_
+
+*test* test *test*
+
+_test_ __test__ _test_
+
+*test* **test** *test*
+
+_a_ _b_
+
+*a* *b*
```

Comment:
Let's add some more tests

 - `\*foo *bar*` -> `<p>*foo <em>bar</em></p>` -> \*foo *bar*
 - `\_foo _bar_` -> `<p>*foo <em>bar</em></p>` -> \_foo _bar_
 - `\*foo *bar* and *baz*` -> `<p>*foo <em>bar</em> and <em>baz</em></p>` -> \*foo *bar* and *baz*
 - `\_foo _bar_ and _baz_` -> `<p>*foo <em>bar</em> and <em>baz</em></p>` -> \_foo _bar_ and _baz_